fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateMTMVInfo.java

```diff
@@ -224,6 +227,15 @@ public void analyzeQuery(ConnectContext ctx) {
         }
         getRelation(planner);
         getColumns(plan);
+        try {
+            // we want to set session variable, so need to new connectContext avoid influence other query
+            statementContext.setConnectContext(
+                    MTMVPlanUtil.createMTMVContext(this.envInfo.getCtlId(), this.envInfo.getDbId()));
```

Comment:
Creating a new ctx here may not be a good method. How about setting environment variables on the ctx of crude oil and setting them back after the method ends? Or there may be other methods instead of using the environment variable setDisableNereidsRules